grasping_benchmarks/graspnet/graspnet_grasp_planner.py

```diff
@@ -15,15 +15,98 @@
 from grasping_benchmarks.base import transformations as tr
 
 import mayavi.mlab as mlab
+import tensorflow as tf
 
 # Import the GraspNet implementation code
 # Requires a GRASPNET_DIR environment variable pointing to the root of the repo
 sys.path.append(os.environ['GRASPNET_DIR'])
 os.chdir(os.environ['GRASPNET_DIR'])
-from demo.main import get_color_for_pc, backproject
+from demo.main import get_color_for_pc, backproject, make_parser
 import grasp_estimator
-import utils.utils as utils
-import utils.visualization_utils as visu_utils
+# import utils as utils
+import visualization_utils 
+
+
+class ArucoBoardData:
+    position = None
+    orientation = None
+    enable_grasp_filter = None
+
+
+def transform_grasp_and_offset_to_world(grasping_pose, offset,camera_pose):
+    """Transform the 6D grasp pose in the world reference frame, given the
+    camera viewpoint
+
+    Parameters
+    ----------
+    grasp_pose : grasp_6d
+        Candidate grasp pose, in the camera ref frame
+    camera_pose : camera_pose
+        Camera pose wrt world ref frame
+
+    Returns
+    -------
+    PoseStamped
+        Candidate grasp pose, in the world reference frame
+    """
+
+    # w_T_cam : camera pose in world ref frame
+    # cam_T_grasp : grasp pose in camera ref frame
+    # w_T_grasp = w_T_cam * cam_T_grasp
+
+    # Construct the 4x4 affine transf matrices from ROS stamped poses
+
+    cam_T_grasp = np.eye(4)
+    cam_T_grasp[:3,:3] = grasping_pose.rotation
+    cam_T_grasp[:3,3] = grasping_pose.position
+
+    grasp_offset = np.eye(4)
+    grasp_offset[:3,3] = offset
+    cam_T_grasp = np.matmul(cam_T_grasp, grasp_offset)
+
+    w_T_cam = np.eye(4)
+    w_T_cam[:3,:3] = camera_pose.extrinsic_params['rotation']
+    w_T_cam[:3,3] = camera_pose.extrinsic_params['position']
+
+    # Obtain the w_T_grasp affine transformation
+    w_T_grasp = np.matmul(w_T_cam, cam_T_grasp)
+
+    
+
+
+    return w_T_grasp   
+
+def check_collision_between_gripper_and_table(grasping_pose,camera_pose, aruco_board_data):
+    
+    offset_distance = 0.01
```

Comment:
Please provide some insight on what this means, e.g. what is the purpose of `offset_distance`, or what gripper are the checkpoints referring to. 

Consider this software package is supposed to be tentatively gripper-agnostic.